.github/workflows/ic.yml

```diff
@@ -0,0 +1,55 @@
+name: Build and Test
+
+on: [push, pull_request]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Install dependencies for Linux
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get -y install bison \
+              autotools-dev \
+              libncurses5-dev \
+              libevent-dev \
+              pkg-config \
+              libutempter-dev \
+              build-essential
+          if [ "${{ matrix.build }}" == "musl" ] || [ "${{ matrix.build }}" == "musl-static" ]; then
+            sudo apt-get -y install musl-dev \
+                        musl-tools
+          fi
+          
+      # Add steps for other operating systems if needed, similar to the Linux steps above.
+
+      - name: Build tmux
+        run: |
+          sh autogen.sh
+          ./configure
+          make
+
+      # Add additional steps for running tests if applicable.
+      - name: Run Tests
+        working-directory: ./regress
+        run: |
+          for script in am-terminal.sh capture-pane-hyperlink.sh combine-test.sh command-order.sh conf-syntax.sh control-client-sanity.sh cursor-test{1..4}.sh has-session-return.sh if-shell-{error,nested,TERM}.sh input-keys.sh kill-session-process-exit.sh new-{session-base-index,window-command}.sh osc-11colours.sh style-trim.sh control-client-size.sh new-session-{no-client,size,command}.sh copy-mode-test-emacs.sh; do
```

Comment:
for test_script in *.sh; do
  chmod +x $test_script
  echo Running test $test_script
  ./$test_script      
done